### ISSUE TITLE
[Usage]: How to set a dynamic temperature when sampling? (decrease gradually as more tokens generated)

### ISSUE THREAD

**Issue by StarDewXXX** [vllm-project/vllm#11276]
### How would you like to use vllm

I want to set a dynamic temperature when sampling. Specifically, I need to dynamically adjust the temperature coefficient based on the length of the currently generated token each time the logits are processed. It seems that `sampling_params` cannot achieve this feature. Which part of the code should I modify?

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by noooop** [comment]
Get the temperature from sampling_params here

https://github.com/vllm-project/vllm/blob/f04e407e6b6b9ce65c16cffda836f05c2ad32682/vllm/model_executor/sampling_metadata.py#L403


but I guess vllm has a caching mechanism.

https://github.com/vllm-project/vllm/blob/f04e407e6b6b9ce65c16cffda836f05c2ad32682/vllm/model_executor/layers/sampler.py#L241C1-L250C1

Modifying such deep code is always hard, good luck
